vyper/parser/events.py

```diff
@@ -0,0 +1,316 @@
+import ast
+
+from vyper.exceptions import (
+    InvalidLiteralException,
+    TypeMismatchException,
+)
+from vyper.parser.expr import (
+    Expr,
+)
+from vyper.parser.lll_node import (
+    LLLnode,
+)
+from vyper.parser.parser_utils import (
+    base_type_conversion,
+    byte_array_to_num,
+    getpos,
+    make_byte_array_copier,
+    make_setter,
+    unwrap_location,
+)
+from vyper.types.types import (
+    BaseType,
+    ByteArrayLike,
+    ListType,
+    get_size_of_type,
+)
+from vyper.utils import (
+    bytes_to_int,
+    ceil32,
+    string_to_bytes,
+)
+
+
+def pack_logging_topics(event_id, args, expected_topics, context, pos):
+    topics = [event_id]
+    code_pos = pos
+    for pos, expected_topic in enumerate(expected_topics):
+        expected_type = expected_topic.typ
+        arg = args[pos]
+        value = Expr(arg, context).lll_node
+        arg_type = value.typ
+
+        if isinstance(arg_type, ByteArrayLike) and isinstance(expected_type, ByteArrayLike):
+            if arg_type.maxlen > expected_type.maxlen:
```

Comment:
probably out of scope of this PR but there is no check here that arg_type and expected_type are the same :)